src/main/java/com/mraof/minestuck/client/gui/SummonerScreen.java

```diff
@@ -0,0 +1,114 @@
+package com.mraof.minestuck.client.gui;
+
+import com.mojang.blaze3d.matrix.MatrixStack;
+import com.mojang.blaze3d.systems.RenderSystem;
+import com.mraof.minestuck.block.redstone.SummonerBlock;
+import com.mraof.minestuck.entity.MSEntityTypes;
+import com.mraof.minestuck.network.MSPacketHandler;
+import com.mraof.minestuck.network.SummonerPacket;
+import com.mraof.minestuck.tileentity.redstone.SummonerTileEntity;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.widget.TextFieldWidget;
+import net.minecraft.client.gui.widget.button.Button;
+import net.minecraft.entity.EntityType;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.text.StringTextComponent;
+import net.minecraftforge.fml.client.gui.widget.ExtendedButton;
+
+import java.util.Optional;
+
+public class SummonerScreen extends Screen
+{
+	private static final ResourceLocation guiBackground = new ResourceLocation("minestuck", "textures/gui/generic_medium.png");
+	
+	private static final int guiWidth = 150;
+	private static final int guiHeight = 98;
+	
+	private final SummonerTileEntity te;
+	private boolean isUntriggerable;
+	private int summonRange;
+	
+	public Button incrementButton;
+	public Button decrementButton;
+	private Button unTriggerableButton;
+	
+	private TextFieldWidget entityTypeTextField;
+	
+	
+	SummonerScreen(SummonerTileEntity te)
+	{
+		super(new StringTextComponent("Summoner"));
+		
+		this.te = te;
+		this.summonRange = te.getSummonRange() > 0 ? te.getSummonRange() : 8; //if its defaulted on creation to 0, set it to the intended default of 8
+		this.isUntriggerable = te.getBlockState().getValue(SummonerBlock.UNTRIGGERABLE);
+	}
+	
+	@Override
+	public void init()
+	{
+		int yOffset = (this.height / 2) - (guiHeight / 2);
+		
+		addButton(incrementButton = new ExtendedButton(this.width / 2 + 20, (height - guiHeight) / 2 + 12, 20, 20, new StringTextComponent("+"), button -> changeRange(1)));
+		addButton(decrementButton = new ExtendedButton(this.width / 2 - 40, (height - guiHeight) / 2 + 12, 20, 20, new StringTextComponent("-"), button -> changeRange(-1)));
+		
+		this.entityTypeTextField = new TextFieldWidget(this.font, this.width / 2 - 60, yOffset + 40, 120, 18, new StringTextComponent("Current Entity Type"));	//TODO Use translation instead, and maybe look at other text fields for what the text should be
+		this.entityTypeTextField.setValue(EntityType.getKey(te.getSummonedEntity()).toString());
+		addButton(entityTypeTextField);
+		
+		if(isUntriggerable)
+			addButton(unTriggerableButton = new ExtendedButton(this.width / 2 - 65, yOffset + 70, 85, 20, new StringTextComponent("UNTRIGGERABLE"), button -> cycleUntriggerable()));
+		else
+			addButton(unTriggerableButton = new ExtendedButton(this.width / 2 - 65, yOffset + 70, 85, 20, new StringTextComponent("TRIGGERABLE"), button -> cycleUntriggerable()));
```

Comment:
I'd suggest simplifying this (and `cycleUntriggerable()`) with a function:
```java
private ITextComponent getTriggerableButtonMessage()
{
	return this.isUntriggerable ? new StringTextComponent("UNTRIGGERABLE") : new StringTextComponent("TRIGGERABLE");
}
```
These button messages could also be candidates for getting translation keys.